insights/parsers/azure_instance_type.py

```diff
@@ -0,0 +1,67 @@
+"""
+AzureInstanceType
+=================
+
+This parser simply reads the output of command
+``curl -H Metadata:true http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2018-10-01&format=text``,
+which is used to check the type of the Azure instance of the host.
+
+For more details, See: https://docs.microsoft.com/en-us/azure/virtual-machines/linux/sizes
+
+"""
+
+from insights.parsers import SkipException, ParseException
+from insights import parser, CommandParser
+from insights.specs import Specs
+
+
+@parser(Specs.azure_instance_type)
+class AzureInstanceType(CommandParser):
+    """
+    Class for parsing the Azure Instance type returned by command
+    ``curl -H Metadata:true http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2018-10-01&format=text``,
+
+
+    Typical output of this command is::
+
+        Standard_L64s_v2
+
+    Raises:
+        SkipException: When content is empty or no parse-able content.
+        ParseException: When type cannot be recognized.
+
+    Attributes:
+        type: The type of VM instance in Azure, e.g: Standard
+        size: The size of VM instance in Azure, e.g: L64s, NC12s
+        version: The version of VM instance in Azure, e.g: v2, v3, `None` for non-version
+        raw: The fully type string returned by the ``curl`` command
```

Comment:
@xiangce your attributes should have a type name following them, like `type (str): ...`.

Reply:
Done. And I fixed the aws_instance_type as well.